internal/ingress/controller/controller.go

```diff
@@ -688,6 +695,15 @@ func (n *NGINXController) createUpstreams(data []*extensions.Ingress, du *ingres
 					upstreams[name].LoadBalancing = anns.LoadBalancing
 				}
 
+				if upstreams[name].SessionAffinity.AffinityType == "" {
+					upstreams[name].SessionAffinity.AffinityType = anns.SessionAffinity.Type
+				}
+
+				if upstreams[name].SessionAffinity.AffinityType == "cookie" {
+					upstreams[name].SessionAffinity.CookieSessionAffinity.Name = anns.SessionAffinity.Cookie.Name
+					upstreams[name].SessionAffinity.CookieSessionAffinity.Hash = anns.SessionAffinity.Cookie.Hash
+				}
+
```

Comment:
Can you extract this controller change into a separate PR and include a corresponding test that fails without this change?